compaction/compaction_manager.cc

```diff
@@ -629,7 +629,7 @@ future<compaction_manager::compaction_stats_opt> compaction_manager::perform_com
 }
 
 std::optional<gate::holder> compaction_manager::start_compaction(table_state& t) {
-    if (_state != state::enabled) {
+    if (_state != state::running || _disabled_state_count > 0) {
```

Comment:
This condition: `(_state != state::running || _disabled_state_count > 0)` is repeated in many places.
How about defining it once, in a predicate method?